reader.go

```diff
@@ -824,6 +828,9 @@ func (r *Reader) run() {
 // ReaderConfig is a configuration object used to create new instances of
 // Reader.
 type ReaderConfig struct {
+	// If the reading should be started from the latest offset instead of the earliest one
+	Latest bool
```

Comment:
"Latest" might not be descriptive enough though. StartFromLatest? ReadFromLatest? Any ideas?

Reply:
Agreed that the option shouldn't be as the first field.

For the naming the issue with just `Offset` is that while it makes sense in the `FirstOffset` and `LastOffset` sense but setting it to anything other than those two wouldn't do what you'd expect it to do when using consumer groups.

Second issue with `Offset` is that in regard of what the current context is the parameter is more like `Whence` instead of `Offset`. (Which then again is more related to the other discussion happening in this same PR)

Reply:
> setting it to anything other than those two wouldn't do what you'd expect it to do when using consumer groups.

I'm not sure I understand.  I'd expect it to start consuming from that offset and continue as normal.

One risk posed when doing this sort of thing is that the consumer can die and restart with the old config.  The I've handled this at $JOB-1 is by running a one off job to reprocess events with since and until offests.  Otherwise we have to figure out the restart story.

I'm not sure what works the best here and I'm open to other ideas, but that seems like a reasonable way forward.

Reply:
The problem is that `offset` only has context in a `topic/partition` pair. This ReaderConfig field was added to be used with a consumer group that handles all partitions in a single topic.

Of course I have a bias here as well since I'm using ReaderConfig for the consumer group while others may use it only to read a single partition for example. In that case it would indeed then work as you'd expect. But in that case setting it to anything other than FirstOffset or LastOffset when using consumer group should probably throw an error.

For generic solution that covers all cases there is the other discussion happening in this same PR. Basically it would mean passing `[partition]offset` map and an optional `whence` which probably would default to absolute value.

Reply:
> The problem is that offset only has context in a topic/partition pair. This ReaderConfig field was added to be used with a consumer group that handles all partitions in a single topic.

Correct.  In my experience operating kafka, it doesn't make a lot of sense to abstract away partition details when using consumer groups.  We might need to make some updates to the implementation in this library to reflect that eventually but we're not using the groups implementation widely internally yet so we haven't hit those cases.